packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -426,6 +440,87 @@ export async function uploadData<T extends UploadDataLookup, K extends keyof T>(
   );
 }
 
+/**
+ * Removes data from the rawData of the largest entity until the overall size
+ * of the data object is less than maxSize (defaulted to UPLOAD_SIZE_MAX).
+ *
+ * @param data
+ */
+export function shrinkRawData<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
```

Comment:
What are your thoughts on making this _not_ generic, so that we can get better typings for `Entity`?

Reply:
Sounds good to me.  I think I just defaulted to that to match the calling function.  In the latest commit it now expects `UploadDataLookup` arrays to allow for either Entities or Relationships (even though I know we're only likely to ever need to shrink Entities)